asllib/doc/ASLTypingReference.tex

```diff
@@ -1618,102 +1797,100 @@ \section{TypingRule.StructuralSubtypeSatisfaction\label{sec:TypingRule.Structura
   \aslto \overname{\Bool}{\vb} \cup \overname{\TTypeError}{\TypeErrorConfig}
 \]
 tests whether a type $\vt$ \emph{structurally-subtype-satisfies} a type $\vs$ is environment $\tenv$,
-returning a type error if one is detected.
+returning the result $\vb$ or a type error, if one is detected.
 
 \subsection{Prose}
-$\vt$ structurally-subtype-satisfies $\vs$ in environment \tenv, if one of the following applies:
+One of the following applies:
+\begin{itemize}
+\item All of the following apply (\textsc{error1}):
   \begin{itemize}
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of an integer type;
-    \item $\vt$ has the underlying type of an integer type.
-    \end{itemize}
-
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of an integer type;
-    \item $\vt$ has the underlying type of an integer type.
-    \end{itemize}
-
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of a string type;
-    \item $\vt$ has the underlying type of a string type.
-    \end{itemize}
+  \item obtaining the \underlyingtype\ of $\vt$ gives a type error;
+  \item the rule results in a type error.
+  \end{itemize}
 
-  \item All of the following apply:
-    \begin{itemize}
-    \item $\vs$ has the underlying type of a boolean type;
-    \item $\vt$ has the underlying type of a boolean type.
+\item All of the following apply (\textsc{error2}):
+  \begin{itemize}
+    \item obtaining the \underlyingtype\ of $\vt$ gives a type $\vttwo$;
+    \item obtaining the \underlyingtype\ of $\vs$ gives a type error;
+    \item the rule results in a type error.
```

Comment:
the rule results in a type error (\textsc{error2}) ?

Reply:
I don't understand this comment.

Reply:
the parenthesized text corresponds to the rule in the Formally paragraph.

Reply:
please would you be able to make that explicit somewhere? maybe in the reading preamble?

Reply:
I explained it in "Formal System" / "Rule Naming"